File: pesocial-frontend/src/services/postService.ts
import api from './api'
import type { CreatePostPayload, Post } from '../types/post'

export const postService = {
  getFeedPosts: async () => {
    const { data } = await api.get<Post[]>('/api/posts')
    return data
  },

  getPostsByAuthor: async (authorId: string) => {
    const { data } = await api.get<Post[]>(`/api/posts/author/${authorId}`)
    return data
  },

  createPost: async (payload: CreatePostPayload) => {
    const { data } = await api.post<Post>('/api/posts', payload)
    return data
  },

  likePost: async (postId: string) => {
    const { data } = await api.patch<Post>(`/api/posts/${postId}/like`)
    return data
  },

  commentPost: async (postId: string, value: string) => {
    const { data } = await api.patch<Post>(`/api/posts/${postId}/comment`, null, {
      params: { value },
    })
    return data
  },

  deletePost: async (postId: string) => {
    await api.delete(`/api/posts/${postId}`)
  },
}

File: src/main/java/com/pesocial/service/PostService.java
package com.pesocial.service;

import java.util.List;

import com.pesocial.dto.post.CreatePostRequest;
import com.pesocial.dto.post.EditPostRequest;
import com.pesocial.dto.post.PostResponseDto;
import com.pesocial.model.post.Post;

public interface PostService {
    Post createPost(CreatePostRequest request);
    List<Post> getFeedPosts(String viewerId);
    Post getPostById(String postId);
    Post editPost(String postId, EditPostRequest request);
    void deletePost(String postId, String userId);
    Post addLike(String postId, String currentUserId);
    Post removeLike(String postId);
    Post addComment(String postId, String comment, String currentUserId);
    Post sharePost(String postId);
    List<Post> getUserPosts(String authorId, String viewerId);
    PostResponseDto toResponse(Post post);
    List<PostResponseDto> toResponseList(List<Post> posts);
}
